src/operator/tensor/ordering_op-inl.h

```diff
@@ -79,6 +80,16 @@ struct TopKParam : public dmlc::Parameter<TopKParam> {
     DMLC_DECLARE_FIELD(is_ascend).set_default(false)
       .describe("Whether to choose k largest or k smallest elements."
                 " Top K largest elements will be chosen if set to false.");
+    DMLC_DECLARE_FIELD(dtype)
+    .add_enum("uint8", mshadow::kUint8)
+    .add_enum("int32", mshadow::kInt32)
+    .add_enum("float16", mshadow::kFloat16)
```

Comment:
should we remove this option given that it won't be supported?

Reply:
OK, I'll remove it.

Reply:
@szha Actually, the dtype of indices can be float16. There is no need to remove the option.